plugins/network-elements/juniper-srx/src/com/cloud/network/resource/JuniperSrxResource.java

```diff
@@ -2078,11 +2078,11 @@ private boolean manageStaticNatRule(SrxCommand command, String publicIp, String
                         xml = replaceXmlValue(xml, "rule-set", _privateZone);
                         xml = replaceXmlValue(xml, "from-zone", _privateZone);
                         xml = replaceXmlValue(xml, "rule-name", ruleName_private);
-                    }
 
-                    if (!sendRequestAndCheckResponse(command, xml, "name", ruleName_private))
-                    {
-                        throw new ExecutionException("Failed to delete trust static NAT rule from public IP " + publicIp + " to private IP " + privateIp);
+                        if (!sendRequestAndCheckResponse(command, xml, "name", ruleName_private))
+                        {
+                            throw new ExecutionException("Failed to delete trust static NAT rule from public IP " + publicIp + " to private IP " + privateIp);
+                        }
```

Comment:
nit- This is one confusing else block :/ I don't think we need the else at all. Essentially it's about sending command and post that checking if private rules delete exists and execute that.